sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -99,6 +105,68 @@
         }
     }
 
+    /**
+     * Asynchronously acquire a token from Active Directory with Aure CLI.
+     *
+     * @param request the details of the token request
+     * @return a Publisher that emits an AccessToken
+     */
+    public Mono<AccessToken> authenticateWithAzureCli(TokenRequestContext request) {
+        String azCommand = "az account get-access-token --output json --resource ";
+
+        StringBuilder command = new StringBuilder();
+        command.append(azCommand);
+        String scopes = ScopeUtil.scopesToResource(request.getScopes());
+        command.append(scopes);
+
+        AccessToken token = null;
+        try {
+            String starter;
+            String switcher;
+            if (System.getProperty("os.name").contains("Windows")) {
+                starter = "cmd.exe";
+                switcher = "/c";
+            } else {
+                starter = "/bin/sh";
+                switcher = "-c";
+            }
+
+            ProcessBuilder builder = new ProcessBuilder(starter, switcher, command.toString());
+            builder.redirectErrorStream(true);
+            Process p = builder.start();
+
+            BufferedReader r = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            String line;
+            StringBuilder output = new StringBuilder();
+            while (true) {
+                line = r.readLine();
+                if (line == null) {
+                    break;
+                }
+                if (line.startsWith("'az' is not recognized") || line.matches("(.*)az:(.*)not found")) {
+                    throw new Exception("Azure CLI not installed");
```

Comment:
this should be 
return Mono.error(logger.logExceptionAsError(new Exception("Azure CL......

Would like to use a better exception class instead of just Exception.